src/inter-aut.cc

```diff
@@ -0,0 +1,410 @@
+            return Mata::FormulaNode{Mata::FormulaNode::Type::OPERAND, token, token,
+                                     Mata::FormulaNode::OperandType::SYM};
+        }
+
+        throw std::runtime_error("Unknown token " + token);
+        assert(false);
+    }
+
+    bool lower_precedence(Mata::FormulaNode::OperatorType op1, Mata::FormulaNode::OperatorType op2) {
+        if (op1 == Mata::FormulaNode::NEG) {
+            return false;
+        } else if (op1 == Mata::FormulaNode::AND && op2 != Mata::FormulaNode::NEG) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Transforms a series of tokes to postfix notation.
+     * @param aut Automaton for which transition formula is parsed
+     * @param tokens Series of tokens representing transition formula parsed from the input text
+     * @return A postfix notation for input
+     */
+    std::vector<Mata::FormulaNode> infix2postfix(Mata::InterAutomaton &aut, const std::vector<std::string> &tokens) {
+        std::vector<Mata::FormulaNode> opstack;
+        std::vector<Mata::FormulaNode> output;
+
+        for (const auto& token : tokens) {
+            Mata::FormulaNode node = create_node(aut, token);
+            switch (node.type) {
+                case Mata::FormulaNode::OPERAND:
+                    output.push_back(node);
+                    break;
+                case Mata::FormulaNode::LEFT_PARENTHESIS:
+                    opstack.push_back(node);
+                    break;
+                case Mata::FormulaNode::RIGHT_PARENTHESIS:
+                    while (!opstack.back().is_leftpar()) {
+                        output.push_back(opstack.back());
+                        opstack.pop_back();
+                        break;
+                    }
+                    opstack.pop_back();
+                    break;
+                case Mata::FormulaNode::OPERATOR:
+                    for (int j = opstack.size()-1; j >= 0; --j) {
+                        assert(!opstack[j].is_operand());
+                        if (opstack[j].is_leftpar())
+                            break;
+                        if (lower_precedence(node.operator_type, opstack[j].operator_type)) {
+                            output.push_back(opstack[j]);
+                            opstack.erase(opstack.begin()+j);
+                        }
+                    }
+                    opstack.push_back(node);
+                    break;
+                default: assert(false);
+            }
+        }
+
+        while (!opstack.empty()) {
+            output.push_back(opstack.back());
+            opstack.pop_back();
+        }
+
+        return output;
+    }
+
+    /**
+     * Create a graph for a postfix representation of transition formula
+     * @param postfix A postfix representation of transition formula
+     * @return A parsed graph
+     */
+    Mata::FormulaGraph postfix2graph(const std::vector<Mata::FormulaNode> &postfix)
```

Comment:
`postfix_to_graph`? I don't see much gain from the `2`

Reply:
you don't need to write two underscores.

Reply:
I generally dislike the use of `2`, `4` etc. in names, too. However, if you feel like the benefits of using them outweigh the lower readability, feel free to use them.

Reply:
renamed